File: server-fixed.js
const http = require('http');
const fs = require('fs');
const path = require('path');
const { exec } = require('child_process');

const PORT = 8000;
const HOSTNAME = 'localhost';
const DEFAULT_HTML_FILE = 'deepseek_html_20250909_2bf8a5.html';

// Mover mimeTypes para fora do handler para melhor performance
const MIME_TYPES = {
    '.html': 'text/html',
    '.js': 'text/javascript',
    '.css': 'text/css',
    '.json': 'application/json',
    '.png': 'image/png',
    '.jpg': 'image/jpg',
    '.gif': 'image/gif',
    '.svg': 'image/svg+xml',
    '.wav': 'audio/wav',
    '.mp4': 'video/mp4',
    '.woff': 'application/font-woff',
    '.ttf': 'application/font-ttf',
    '.eot': 'application/vnd.ms-fontobject',
    '.otf': 'application/font-otf',
    '.wasm': 'application/wasm'
};

// Função para validar e sanitizar o caminho do arquivo
function sanitizePath(requestPath) {
    // Normalizar o caminho
    let filePath = path.normalize('.' + requestPath);
    
    // Verificar se é o diretório raiz
    if (filePath === './' || filePath === '.') {
        filePath = './' + DEFAULT_HTML_FILE;
    }
    
    // Verificar se o caminho está dentro do diretório atual (prevenir path traversal)
    const resolvedPath = path.resolve(filePath);
    const currentDir = path.resolve('.');
    
    if (!resolvedPath.startsWith(currentDir)) {
        return null; // Caminho inválido
    }
    
    return filePath;
}

const server = http.createServer((req, res) => {
    const filePath = sanitizePath(req.url);
    
    if (!filePath) {
        res.writeHead(403, { 'Content-Type': 'text/html' });
        res.end('<h1>403 - Acesso negado</h1>', 'utf-8');
        return;
    }

    const extname = String(path.extname(filePath)).toLowerCase();
    const contentType = MIME_TYPES[extname] || 'application/octet-stream';

    fs.readFile(filePath, (error, content) => {
        if (error) {
            if (error.code === 'ENOENT') {
                res.writeHead(404, { 'Content-Type': 'text/html' });
                res.end('<h1>404 - Arquivo não encontrado</h1>', 'utf-8');
            } else {
                console.error('Erro interno do servidor:', error);
                res.writeHead(500);
                res.end('Erro interno do servidor\n');
            }
        } else {
            res.writeHead(200, { 'Content-Type': contentType });
            res.end(content, 'utf-8');
        }
    });
});

server.listen(PORT, HOSTNAME, () => {
    console.log(`Servidor rodando em: http://${HOSTNAME}:${PORT}`);
    console.log(`Acesse o sistema em: http://${HOSTNAME}:${PORT}/${DEFAULT_HTML_FILE}`);
    console.log('Pressione Ctrl+C para parar o servidor');
    
    // Abre automaticamente no navegador com tratamento de erro
    exec(`start http://${HOSTNAME}:${PORT}/${DEFAULT_HTML_FILE}`, (error) => {
        if (error) {
            console.log('Não foi possível abrir o navegador automaticamente.');
            console.log('Acesse manualmente:', `http://${HOSTNAME}:${PORT}/${DEFAULT_HTML_FILE}`);
        }
    });
});

// Tratamento gracioso de encerramento
process.on('SIGINT', () => {
    console.log('\nEncerrando servidor...');
    server.close(() => {
        console.log('Servidor encerrado.');
        process.exit(0);
    });
});

File: server-fixed.py
#!/usr/bin/env python3
import http.server
import socketserver
import webbrowser
import os
import sys
from pathlib import Path

PORT = 8000
DEFAULT_HTML_FILE = 'deepseek_html_20250909_2bf8a5.html'

# Usar diretório atual em vez de caminho hardcoded
DIRECTORY = os.getcwd()

class Handler(http.server.SimpleHTTPRequestHandler):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, directory=DIRECTORY, **kwargs)
    
    def do_GET(self):
        # Redirecionar raiz para o arquivo HTML principal
        if self.path == '/':
            self.path = '/' + DEFAULT_HTML_FILE
        super().do_GET()

def main():
    try:
        # Verificar se o arquivo HTML existe
        html_file_path = Path(DIRECTORY) / DEFAULT_HTML_FILE
        if not html_file_path.exists():
            print(f"Erro: Arquivo {DEFAULT_HTML_FILE} não encontrado no diretório atual.")
            print(f"Diretório atual: {DIRECTORY}")
            sys.exit(1)
        
        with socketserver.TCPServer(("", PORT), Handler) as httpd:
            print(f"Servidor rodando em: http://localhost:{PORT}")
            print(f"Acesse o sistema em: http://localhost:{PORT}/{DEFAULT_HTML_FILE}")
            print("Pressione Ctrl+C para parar o servidor")
            
            # Tentar abrir automaticamente no navegador
            try:
                webbrowser.open(f"http://localhost:{PORT}/{DEFAULT_HTML_FILE}")
            except Exception as e:
                print(f"Não foi possível abrir o navegador automaticamente: {e}")
                print(f"Acesse manualmente: http://localhost:{PORT}/{DEFAULT_HTML_FILE}")
            
            try:
                httpd.serve_forever()
            except KeyboardInterrupt:
                print("\nServidor parado.")
                
    except OSError as e:
        if e.errno == 98:  # Address already in use
            print(f"Erro: Porta {PORT} já está em uso.")
            print("Tente usar uma porta diferente ou pare o processo que está usando esta porta.")
        else:
            print(f"Erro ao iniciar servidor: {e}")
        sys.exit(1)
    except Exception as e:
        print(f"Erro inesperado: {e}")
        sys.exit(1)

if __name__ == "__main__":
    main()